### PR TITLE
Various improvements

### PR DESCRIPTION
**Pull request by infinisil**
- Remove unnecessary **Description** headers
- Rename **Rationale and Alternatives** to just **Alternatives**
- Insert must/may/should more diligently
- Add some TODOs where things are unclear
- Remove numberings from examples when not needed
- Minor clarity improvements and simplifications throughout

### CODE REVIEW

File: rfcs/0166-nix-formatting.md [hunk 1]
```diff
@@ -262,34 +262,34 @@ Terms and definitions:
 - Braces: `{}`
 - Parentheses: `()`
 
-- Line breaks may not be preserved, but empty lines must always be preserved.
-  - This allows the formatter to compact down multi-line expressions if necessary, while still allowing to structure the code appropriately.
-  - Multiple subsequent empty lines must get collapsed into a single one.
+- Line breaks may be added or removed. Consecutive empty lines must be turned into a single empty line.
```
Comment (rhendric) [inline rfcs/0166-nix-formatting.md:265]: Sick of me yet? :sweat_smile: Consecutive empty lines implies more than one, I think, so this seems to allow a formatter to remove a single empty line if it chooses to, or insert new single empty lines, which I suspect is not the intent.

```suggestion
- Line breaks may be added or removed, but empty lines must not be created. Single empty lines must be preserved, and consecutive empty lines must be turned into a single empty line.
```

File: rfcs/0166-nix-formatting.md [hunk 2]
```diff
@@ -398,31 +421,28 @@ These limits should be configurable, and the soft length limit should default to
...
 
 ### Expansion of expressions
 
-**Description**
-
-Unless stated otherwise, any expression that fits onto one single line will be trivially formatted as such.
+Unless stated otherwise, any expression that fits onto one single line must be trivially formatted as such.
 
-For sequences of items in expressions, like elements in a list, key-value pairs in attribute sets, or function arguments, the following applies:
+For list elements, attributes and function arguments, the following applies:
```
Comment (rhendric) [inline rfcs/0166-nix-formatting.md:433]: An Oxford comma here would prevent someone from thinking ‘list elements’ is a concept that expands to ‘attributes and function arguments’.
```suggestion
For list elements, attributes, and function arguments, the following applies:
```

File: rfcs/0166-nix-formatting.md (hunk 1 above)
Comment (infinisil) [reply rfcs/0166-nix-formatting.md:265]: Sounds good to me!

File: rfcs/0166-nix-formatting.md (hunk 2 above)
Comment (infinisil) [reply rfcs/0166-nix-formatting.md:433]: Sounds good!

**Review by tomberek — APPROVED** [review]
Most of this was discussed and iterated on at last meeting.

**Pull request merged**